modular_skyrat/modules/microfusion/code/microfusion_gun_attachments.dm

```diff
@@ -0,0 +1,446 @@
+/obj/item/microfusion_gun_attachment/proc/process_fire(obj/item/gun/microfusion/microfusion_gun, obj/item/ammo_casing/chambered)
+	return
+
+/obj/item/microfusion_gun_attachment/proc/remove_attachment(obj/item/gun/microfusion/microfusion_gun)
+	SHOULD_CALL_PARENT(TRUE)
+	microfusion_gun.heat_per_shot -= heat_addition
+	microfusion_gun.update_appearance()
+	microfusion_gun.extra_power_usage -= power_usage
+	return
+
+/*
+Returns a list of modifications of this attachment, it must return a list within a list list(list()).
+All of the following must be returned.
+list(list("title" = "Toggle [toggle ? "OFF" : "ON"]", "icon" = "power-off", "color" = "blue" "reference" = "toggle_on_off"))
+title - The title of the modification button
+icon - The icon of the modification button
+color - The color of the modification button
+reference - The reference of the modification button, this is used to call the proc when the run modify data proc is called.
+*/
+/obj/item/microfusion_gun_attachment/proc/get_modify_data()
+	return
+
+/obj/item/microfusion_gun_attachment/proc/run_modify_data(params, mob/living/user, obj/item/gun/microfusion/microfusion_gun)
+	return
+
+/obj/item/microfusion_gun_attachment/proc/get_information_data()
+	return
+
+/*
+SCATTER ATTACHMENT
+
+The cell is stable and will not emit sparks when firing.
+*/
+/obj/item/microfusion_gun_attachment/scatter
+	name = "diffuser microfusion lens upgrade"
+	desc = "A diffusing lens system capable of splitting one beam into three. However, the additional ionizing of the air will cause higher recoil."
+	icon_state = "attachment_scatter"
+	attachment_overlay_icon_state = "attachment_scatter"
+	slot = GUN_SLOT_BARREL
+	/// How many pellets are we going to add to the existing amount on the gun?
+	var/pellets_to_add = 2
+	/// The variation in pellet scatter.
+	var/variance_to_add = 20
+	/// How much recoil are we adding?
+	var/recoil_to_add = 1
+	/// The spread to add.
+	var/spread_to_add = 10
+
+/obj/item/microfusion_gun_attachment/scatter/run_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.recoil += recoil_to_add
+	microfusion_gun.spread += spread_to_add
+	microfusion_gun.microfusion_lens.pellets += pellets_to_add
+	microfusion_gun.microfusion_lens.variance += variance_to_add
+
+/obj/item/microfusion_gun_attachment/scatter/process_fire(obj/item/gun/microfusion/microfusion_gun, obj/item/ammo_casing/chambered)
+	. = ..()
+	chambered.loaded_projectile?.damage = chambered.loaded_projectile.damage / chambered.pellets
+
+/obj/item/microfusion_gun_attachment/scatter/remove_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.recoil -= recoil_to_add
+	microfusion_gun.spread -= spread_to_add
+	microfusion_gun.microfusion_lens.pellets -= microfusion_gun.microfusion_lens.pellets
+	microfusion_gun.microfusion_lens.variance -= microfusion_gun.microfusion_lens.variance
+
+/*
+REPEATER ATTACHMENT
+
+The gun can fire volleys of shots.
+*/
```

Comment:
Do the comment blocks like this, using the auto-commentblock extension I just sent you ;)
```suggestion
/**
 * REPEATER ATTACHMENT
 *
 * The gun can fire volleys of shots.
 */
```